src/main/scala/com/github/tmnd1991/spark/testing/SparkFunSuite.scala

```diff
@@ -17,19 +17,37 @@
 
 package com.github.tmnd1991.spark.testing
 
-import java.io.File
+import com.github.tmnd1991.spark.testing.SparkFunSuite.warehouseLocation
 
+import java.io.File
 import org.apache.spark.Bridge
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.SparkSession
-import org.scalatest.{ BeforeAndAfterAll, FunSuite, Outcome }
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Outcome}
 
 /**
  * Base abstract class for all unit tests in Spark for handling common functionality.
  */
 abstract class SparkFunSuite extends FunSuite with BeforeAndAfterAll with Logging {
-
-  lazy val spark: SparkSession = SparkFunSuite.spark.newSession()
+  lazy val spark: SparkSession = {
```

Comment:
if sparkSession is a field of the SparkFunSuite, it will try to create a new instance for every test, but the new instance won't be created for every test (`getOrCreate` will `create` the first time and `get` all the subsequent calls), but it will register multiple shutdown hooks that will all try to stop the **same** spark session multiple times.
Also the fact that the call to `newSession` is skipped now, all the tests will share the same session, with also all the changes to configurations that might be made by other tests (i.e. having the side-effect that tests will interfere with each other more than it is now, causing some execution order related issues).

---

Bottom line: I agree on the new confs you are passing to start the test context, but I would like to keep the spark session inside the singleton (please also set spark.testing = true).